models/db/setting.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package db
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"code.gitea.io/gitea/modules/cache"
+	setting_module "code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/timeutil"
+	"code.gitea.io/gitea/modules/util"
+
+	"xorm.io/builder"
+)
+
+type ResourceSetting struct {
+	ID           int64              `xorm:"pk autoincr"`
+	GroupID      int64              `xorm:"index unique(key_repoid)"`               // to load all of some group's settings
+	SettingKey   string             `xorm:"varchar(255) index unique(key_groupid)"` // ensure key is always lowercase
+	SettingValue string             `xorm:"text"`
+	Version      int                `xorm:"version"` // prevent to override
+	Created      timeutil.TimeStamp `xorm:"created"`
+	Updated      timeutil.TimeStamp `xorm:"updated"`
+}
+
+func (s *ResourceSetting) AsBool() bool {
+	if s == nil {
+		return false
+	}
+
+	b, _ := strconv.ParseBool(s.SettingValue)
+	return b
+}
+
+// GetSettings returns specific settings
+func GetSettings(ctx context.Context, tableName string, groupID int64, keys []string) (map[string]*ResourceSetting, error) {
+	for i := 0; i < len(keys); i++ {
+		keys[i] = strings.ToLower(keys[i])
+	}
+	settings := make([]*ResourceSetting, 0, len(keys))
+	if err := GetEngine(ctx).
+		Table(tableName).
+		Where("group_id=?", groupID).
+		And(builder.In("setting_key", keys)).
+		Find(&settings); err != nil {
+		return nil, err
+	}
+	settingsMap := make(map[string]*ResourceSetting)
+	for _, s := range settings {
+		settingsMap[s.SettingKey] = s
+	}
+	return settingsMap, nil
+}
+
+func ValidateSettingKey(key string) error {
+	if len(key) == 0 {
+		return fmt.Errorf("setting key must be set")
+	}
+	if strings.ToLower(key) != key {
+		return fmt.Errorf("setting key should be lowercase")
+	}
+	return nil
+}
+
+// genSettingCacheKey returns the cache key for some configuration
+func genSettingCacheKey(tableName string, groupID int64, key string) string {
+	return fmt.Sprintf("%s.setting.%d.%s", tableName, groupID, key)
+}
+
+// GetSettingNoCache returns specific setting without using the cache
+func GetSettingNoCache(ctx context.Context, tableName string, groupID int64, key string) (*ResourceSetting, error) {
+	v, err := GetSettings(ctx, tableName, groupID, []string{key})
+	if err != nil {
+		return nil, err
+	}
+	if len(v) == 0 {
+		return nil, fmt.Errorf("%s[%d] setting[%s]: %w", tableName, groupID, key, util.ErrNotExist)
```

Comment:
I think the `setting` is redundant (`repo_setting`, `user_setting`)
```suggestion
		return nil, fmt.Errorf("%s[%d][%s]: %w", tableName, groupID, key, util.ErrNotExist)
```